sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -1991,6 +1991,22 @@ object functions {
   def tan(columnName: String): Column = tan(Column(columnName))
 
   /**
+   * Computes the cotangent of the given column.
+   *
+   * @group math_funcs
+   * @since 2.3.0
+   */
+  def cot(e: Column): Column = withExpr { Cot(e.expr) }
+
+  /**
+   * Computes the cotangent of the given column name.
+   *
+   * @group math_funcs
+   * @since 2.3.0
+   */
+  def cot(colName: String): Column = cot(Column(colName))
+
+  /**
```

Comment:
We do not add these functions here in this pr. See: https://github.com/apache/spark/pull/17999#discussion-diff-116842201R2007